src/manifolds/PowerManifold.jl

```diff
@@ -55,50 +55,56 @@ struct PowerFVectorDistribution{
 end
 
 const PowerManifoldMultidimensional =
-    AbstractPowerManifold{𝔽,<:Manifold{𝔽},ArrayPowerRepresentation} where {𝔽}
+    AbstractPowerManifold{𝔽,<:AbstractManifold{𝔽},ArrayPowerRepresentation} where {𝔽}
 
-Base.:^(M::Manifold, n) = PowerManifold(M, n...)
+Base.:^(M::AbstractManifold, n) = PowerManifold(M, n...)
 
-function allocate_result(M::PowerManifoldNested, f::typeof(flat), w::TFVector, x)
-    alloc = [allocate(_access_nested(w.data, i)) for i in get_iterator(M)]
-    return FVector(CotangentSpace, alloc)
+function allocate_result(M::PowerManifoldNested, f::typeof(get_point), x)
+    return [allocate_result(M.manifold, f, _access_nested(x, i)) for i in get_iterator(M)]
 end
-function allocate_result(M::PowerManifoldNested, f::typeof(sharp), w::CoTFVector, x)
-    alloc = [allocate(_access_nested(w.data, i)) for i in get_iterator(M)]
-    return FVector(TangentSpace, alloc)
+function allocate_result(M::PowerManifoldNested, f::typeof(get_point_coordinates), p)
+    return invoke(
+        allocate_result,
+        Tuple{AbstractManifold,typeof(get_point_coordinates),Any},
+        M,
+        f,
+        p,
+    )
 end
 
 default_metric_dispatch(::AbstractPowerManifold, ::PowerMetric) = Val(true)
 
 function det_local_metric(
     M::MetricManifold{PowerMetric,𝔽,<:AbstractPowerManifold{𝔽}},
     p::AbstractArray,
+    B::DefaultOrthonormalBasis,
 ) where {𝔽}
     result = one(number_eltype(p))
     rep_size = representation_size(M.manifold)
     for i in get_iterator(M)
-        result *= det_local_metric(M.manifold, _read(M, rep_size, p, i))
+        result *= det_local_metric(M.manifold, _read(M, rep_size, p, i), B)
```

Comment:
When I tried to write a test for this I encountered two problems
(1) `PowerMetric` is the default metric for an `AbstractPowerManifold` so by transparency rules this seems to pass down to Metricmanifold.jl line 346
(2) I think this line has to have (if we reach it after solving (1)) `M.manifold.manifold`, since `M` is a metric manifold, then `M.manifold is the power manifold and we need its `.manifold` here, actually.

Reply:
(1) I guess we should have this method for `AbstractPowerManifold` directly instead of `MetricManifold`? Or maybe for both?
(2) Yes, that seems right.

Reply:
With abstract, the transparency still does the same, I think? Can you try?

Reply:
OK, I'll try that.